### ISSUE TITLE
Phpdoc rules

### ISSUE THREAD

**Issue by thomasvargiu** [facile-it/facile-coding-standard#6]
Should be possibile to define rulesets for phpdoc.

Suggested rules for phpdoc:

```php
$phpdocRules = [
    'align_multiline_comment' => true,
    'no_blank_lines_after_phpdoc' => true,
    'phpdoc_align' => true,
    'phpdoc_annotation_without_dot' => true,
    'phpdoc_indent' => true,
    'phpdoc_no_package' => true,
    'phpdoc_no_useless_inheritdoc' => true,
    'phpdoc_order' => true,
    'phpdoc_scalar' => true,
    'phpdoc_separation' => true,
    'phpdoc_single_line_var_spacing' => true,
    'phpdoc_to_comment' => true,
    'phpdoc_trim' => true,
    'phpdoc_types' => true,
    'phpdoc_var_without_name' => true,
];

```

**Comment by Jean85** [comment]
I'm not sure about the `phpdoc_align` and the `phpdoc_separation` rules.. But it's a personal preference!

**Comment by Jean85** [comment]
Since we're breaking BC with #10 and #11, should we add those too?
I'm strongly against `phpdoc_align` BTW, which is this one:
```diff
<?php
/**
- * @param  EngineInterface $templating
- * @param string      $format
- * @param  int  $code       an HTTP response status code
- * @param    bool         $debug
- * @param  mixed    &$reference     a parameter passed by reference
+ * @param EngineInterface $templating
+ * @param string          $format
+ * @param int             $code       an HTTP response status code
+ * @param bool            $debug
+ * @param mixed           &$reference a parameter passed by reference
 */
```

**Comment by alexmanno** [comment]
@Jean85 IMHO `phpdoc_align` make phpdoc more readable. Can you explain why you're against that?

**Comment by gianiaz** [comment]
@alexmanno personally I'm against phpdoc_align too.
if one of the types has a very long name I don't think that allignement gives more readability.

**Comment by thomasvargiu** [comment]
I agree, `phpdoc_align` should be disabled. It's a mess when just one parameter has a long type class.
We can also remove `phpdoc_separation`, I think it's not strongly needed.

**Comment by Jean85** [comment]
I do not use `phpdoc_separation`, and I'm not so strongly attached to it, so we can drop it if you want. As for other rules, I've scanned them while writing #13 and I didn't find anything else to add.

**Comment by Jean85** [comment]
> @Jean85 IMHO phpdoc_align make phpdoc more readable. Can you explain why you're against that?

@alexmanno I'll give you a practical example, using the same piece of code of above. Let's say that I rename `EngineInterface` to `Engine` (or vice versa, it's the same). This would be the diff with the fixer:

```diff
<?php
/**
- * @param EngineInterface $templating
- * @param string          $format
- * @param int             $code       an HTTP response status code
- * @param bool            $debug
- * @param mixed           &$reference a parameter passed by reference
+ * @param Engine $templating
+ * @param string $format
+ * @param int    $code       an HTTP response status code
+ * @param bool   $debug
+ * @param mixed  &$reference a parameter passed by reference
 */
```
The diff is huge for no real reason, and it's not even more readable, because the column is too tight.

**Comment by thomasvargiu** [comment]
Implemented in #16
